src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxy.cs

```diff
@@ -39,5 +39,16 @@ protected DispatchProxy()
         {
             return (T)DispatchProxyGenerator.CreateProxyInstance(typeof(TProxy), typeof(T));
         }
+
+        [RequiresDynamicCode("Creating a proxy instance requires generating code at runtime")]
+        public static object Create([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type interfaceType, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type proxyType)
+        {
+            if (!proxyType.IsAssignableTo(typeof(DispatchProxy)))
+            {
+                throw new ArgumentException("proxyType must be assignable to DispatchProxy");
```

Comment:
This message should be defined in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.DispatchProxy/src/Resources/Strings.resx

Reply:
Updated. The new exception message looks like so:  
**System.ArgumentException**: ` The provided proxy type '<YOUR_TYPE>' is not assignable to the type 'DispatchProxy'.`